src/modules/verification-service/verification-service.controller.ts

```diff
@@ -1,19 +1,18 @@
 import { Body, Controller, Post } from '@nestjs/common';
-import { IVerificationService } from '../../interfaces/service/IVerificationService';
+import { VerificationService } from 'src/modules/verification-service/verification.service';
 import { VerifyDocumentDTO } from '../../dtos/verify-document.dto';
-import { ProviderFactory } from '../../factories/provider.factory';
-import { IMakeServiceType } from '../../interfaces/factory/IMakeServiceType';
 import { ServiceType } from '../../enums/service-type.enum';
+import { ProviderFactory } from '../../factories/provider.factory';
 
 @Controller()
 export class VerificationServiceController {
   constructor(
     private readonly providerFactory: ProviderFactory,
-    private readonly verificationService: IVerificationService,
+    private readonly verificationService: VerificationService,
   ) {
     this.verificationService = this.providerFactory.makeService({
       serviceType: ServiceType.VERIFICATION_SERVICE_SERVICE,
-    } as IMakeServiceType);
+    }) as VerificationService;
```

Comment:
Avoiding this concrete implementation is the reason why I used those interfaces instead. This is technically violating the dependency inversion principle when compared to an implementation that causes both the controller and service impl. to depend on the same interface which in this case is IVerificationService